--- ghidra_12.1.2_PUBLIC/support/launch.sh.orig
+++ ghidra_12.1.2_PUBLIC/support/launch.sh
@@ -172,16 +172,18 @@
 # Get the configurable environment variables from the launch properties
 # Only set them if they are currently unset or empty
 while IFS=$'\r\n' read -r line; do
+	[ -z "$line" ] && continue
 	IFS='=' read -r key value <<< "$line"
 	if [ -z ${!key} ]; then
 		export $key=$value
 	fi
-done < <("${JAVA_CMD}" -cp "${LS_CPATH}" LaunchSupport "${INSTALL_DIR}" -envvars)
+done <<< "$("${JAVA_CMD}" -cp "${LS_CPATH}" LaunchSupport "${INSTALL_DIR}" -envvars)"
 
 # Get the configurable VM arguments from the launch properties
 while IFS=$'\r\n' read -r line; do
+	[ -z "$line" ] && continue
 	VMARGS_FROM_LAUNCH_PROPS+=("$line")
-done < <("${JAVA_CMD}" -cp "${LS_CPATH}" LaunchSupport "${INSTALL_DIR}" -vmargs)
+done <<< "$("${JAVA_CMD}" -cp "${LS_CPATH}" LaunchSupport "${INSTALL_DIR}" -vmargs)"
 
 # Add extra macOS VM arguments
 if [ "$(uname -s)" = "Darwin" ]; then
@@ -238,4 +240,3 @@
 	set -o noglob; "${JAVA_CMD}" "${VMARGS_FROM_LAUNCH_PROPS[@]}" "${VMARGS_FROM_LAUNCH_SH[@]}" ${VMARGS_FROM_CALLER} -showversion -cp "${CPATH}" ghidra.Ghidra ${CLASSNAME} "${ARGS[@]}"
 	exit $?
 fi
-
